Digitalshadows new playbooks

ds-jdcockrill opened the pull request:
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - #18246 has now been merged

## Related Issues
N/A

## Description
Add playbooks demonstrating:
- Enrichment of Digital Shadows SearchLight incidents
- Enrichment of IoCs such as URL, Domain, file hash using Digital Shadows functions within Cortex XSOAR

## Screenshots
N/A

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

kgal-pan commented:
> Regarding [#18249 (comment)](https://github.com/demisto/content/pull/18249#issuecomment-1075333346). The failure is due to there being multiple new release notes in this PR.
> 
> This is likely because this PR is on hold behind #18246 and has brought in the release note from that PR as well as the release note for this one. Once the other PR is merged this ought to go away.

I believe you can actually remove `1_0_3.md` since it's going to be added when you sync the upstream with `master`.

kgal-pan commented on Packs/DigitalShadows/pack_metadata.json:
```diff
@@ -1,20 +1,67 @@
+    "name": "Digital Shadows SearchLight",
+    "description": "Digital Shadows monitors and manages an organization's digital risk across the open, deep, and dark web.",
+    "support": "partner",
+    "currentVersion": "1.0.4",
```
bump this down to `1.0.3`

kgal-pan commented on Packs/DigitalShadows/ReleaseNotes/1_0_4.md:
```diff
@@ -0,0 +1,22 @@
+
```
```bash
rm Packs/DigitalShadows/ReleaseNotes/1_0_3.md
mv Packs/DigitalShadows/ReleaseNotes/1_0_4.md Packs/DigitalShadows/ReleaseNotes/1_0_3.md
```

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=18249) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=18249) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: ds-jdcockrill<br/>:white_check_mark: rossbowerman<br/>:x: Jamie Cockrill<br/><hr/>**Jamie Cockrill** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=18249) it.</sub>

ds-jdcockrill commented:
> > Regarding [#18249 (comment)](https://github.com/demisto/content/pull/18249#issuecomment-1075333346). The failure is due to there being multiple new release notes in this PR.
> > This is likely because this PR is on hold behind #18246 and has brought in the release note from that PR as well as the release note for this one. Once the other PR is merged this ought to go away.
> 
> ~I believe you can actually remove `1_0_3.md` since it's going to be added when you sync the upstream with `master` (once #18256 is merged)~
> 
> **UPDATE**
> 
> Looks like the file was actually deleted in [`a70c7e8` (#18256)](https://github.com/demisto/content/pull/18256/commits/a70c7e88c8b8b303a5736574c98d9d0544d69d4a)



> > Regarding [#18249 (comment)](https://github.com/demisto/content/pull/18249#issuecomment-1075333346). The failure is due to there being multiple new release notes in this PR.
> > This is likely because this PR is on hold behind #18246 and has brought in the release note from that PR as well as the release note for this one. Once the other PR is merged this ought to go away.
> 
> ~I believe you can actually remove `1_0_3.md` since it's going to be added when you sync the upstream with `master` (once #18256 is merged)~
> 
> **UPDATE**
> 
> Looks like the file was actually deleted in [`a70c7e8` (#18256)](https://github.com/demisto/content/pull/18256/commits/a70c7e88c8b8b303a5736574c98d9d0544d69d4a)

@kgal-pan - I've changed the version number as advised and renamed the release notes file from `1_0_4` to `1_0_3`. There might end up being some merge conflicts from the contrib branch into master since I've essentially repeated what @yucohen did in https://github.com/demisto/content/pull/18256/commits/a70c7e88c8b8b303a5736574c98d9d0544d69d4a to downversion and remove the `1_0_3` release note I'd created in the other PR.

Hopefully this is done.

ds-jdcockrill commented:
@kgal-pan - the CLAassistant is complaining again about an unfortunate commit (09ed85d545915b177ad081947f06d63f20d4db85) which has ended up in the history again. The [PR guidelines](https://xsoar.pan.dev/docs/contributing/conventions) say to not force-push to rewrite the history - what do you suggest?

Feel free to reach out on slack and I'll happily resolve as suggested.

ds-jdcockrill commented:
Comments from review with @tomer-pan , @yucohen & @kgal-pan:

- consider adding a custom incident type for Digital Shadows incidents and a mapper into the XSOAR incident fields
- consider adding a custom layout to better present the Digital Shadows incident data
- prefer '!SetAndHandleEmpty' rather than extend as that can help avoid errors associated with no-results
- fix `ds-search - DOMAIN on Whois` step in  `Digital_Shadows_-_Domain_Alert_Intelligence_(Automated).yml` - currently the output is not handled
- remove `Packs/DigitalShadows/Playbooks)/Digital_Shadows_-_CVE_Alert_Intelligence_and_Reputation_(Automated).yml`

kgal-pan commented:
@ds-jdcockrill - Can you please use `Utils/git_pull_master_into_fork.sh` to fix the validation build failures?

tomer-pan commented on Packs/DigitalShadows/Layouts/layoutscontainer-Digital_Shadows_Intelligence_Incident_Layout.json:
```diff
@@ -0,0 +1,334 @@
+{
```
@ds-jdcockrill Why do we need another layout? They both look the same.

ds-jdcockrill replied:
They are slightly different - the Client incident doesn't have 'Impact' or 'Mitigation' data and so the central panel in the layout omits those fields.

tomer-pan replied:
I see. It seems like a tiny difference between the 2. Also, in case they are empty , they would not show up in the layout (unless the user click on "show empty fields") - seems unneeded to me.
BTW, you wrote that the "Client" incident doesn't have these fields but I can see that it is the Client layout and not the intelligence layout that contain these fields, so please make sure you didn't confuse between them.

ds-jdcockrill replied:
Hi Tomer,

You are correct, Client Incident is the one that has 'Impact' and 'Mitigation' on it - this is the correct way round and it's just my comment that is incorrect. Having thought about it, I'd like to keep the layouts separate as it is probable that in a subsequent update we will do a bit more work on the layouts and playbooks associated with each. The intelligence incident will more likely end up looking like a threat intelligence report that can be processed in slow-time whereas the client incident is more of a real-time threat that needs investigating and addressing.

tomer-pan commented on Packs/DigitalShadows/IncidentTypes/incidenttype-Digital_Shadows_Intelligence_Incident.json:
```diff
@@ -0,0 +1,26 @@
+{
```
@ds-jdcockrill Why do we need 2 types if the same default playbook is associated?
Also, In case you keep only one of the types, you can also remove the classifier (not the mapper) since the type can be configured through the integration instance config.

ds-jdcockrill replied:
The two types have different fields associated with them - they also mean different things to a user.

- Intelligence incidents are general intelligence information about some threat actor or event
- Client incidents are specific to a particular customer, for instance one of their websites is being impersonated for Phishing purposes, or similar.

They can also contain a slightly different structure (mostly the Impact and Mitigation fields) and so I separated them.

tomer-pan replied:
Okay, I will leave this decision to you, if you feel it is needed. Please see my latest comment about the layouts - I still believe it can be done with a single layout but you can choose whether to keep the 2 or not.

ds-jdcockrill replied:
Thanks - I'll keep these the same and think about the layout.

yucohen reviewed (approved):
Great work!
